### PR TITLE
PLANET-2109 enpages test

### PR DESCRIPTION
**Pull request by angtheod**
- Improved/Enhanced the Engaging Networks selenium test by adding an extra check for the 'Edit in EN' Action located inside the EN pages table that is retrieved by the EN plugin. This check tests whether the Edit action actually navigates the user to the correct page within the Engaging Networks platform.
- Fixed some of the catch blocks so that they produce correct messages in order to inform better about possible Failures.
- Used cssSelector() instead of xpath/className in some cases because it is faster.
- Moved the assertElementNotFound method to the parent abstract class and updated it to work better.
- Changed the way that the P4_login class logs into WP panel so that it works also when running the tests on Localhost.

Update after code review:
- Added a more adjustable 'waiting' method which requires less work when writing a test and makes tests faster since it can wait until an element is displayed instead of waiting a hard coded amount of time.
- Moved configuration loading into AbstractClass.
- Updated so that we can test with Firefox if we set 'firefox' as browser in the configuration file
(*Note Firefox testing fails currently due to an issue of Geckodriver with latest Selenium versions, see https://github.com/facebook/php-webdriver/issues/469 )

### CODE REVIEW

File: tests/p4/ENPages.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	/**
+	 * @param array $args Associative array with values needed.
+	 */
+	protected function checkNavigationToEN( $args ) {
+		// Get all opened browser tabs.
+		$tabs = $this->_driver->getWindowHandles();
+		// Switch to the newly opened EN platform tab.
+		$this->_driver->switchTo()->window( end( $tabs ) );
+
+		sleep( 5 );
```
Comment (kirdia) [inline tests/p4/ENPages.php]: @angtheod I know we have used hardcoded intervals in many places, but I think we should switch to explicit waits that webdriver offers. This interval fails for me locally.

I tried this
`$this->_driver->wait(10, 500)->until(
    WebDriverExpectedCondition::visibilityOfElementLocated(WebDriverBy::id('enLoginUsername'))
);`

Also we can wrap it inside a trycatch block for `TimeOutException`.

**Comment by angtheod** [comment]
@kirdia please check if this is what you meant

File: tests/AbstractClass.php [hunk 2]
```diff
@@ -1,12 +1,15 @@
 <?php
 
-
 include 'P4_Functions.php';
 
 abstract class AbstractClass extends PHPUnit\Framework\TestCase {
```
Comment (kirdia) [inline tests/AbstractClass.php]: Indentation seems off in this file

**Comment by kirdia** [comment]
@angtheod Yes, it is nice 👍 only some indentation fix needed

**Pull request merged**